File: Examples/test-suite/perl5/cpp_enum_runme.pl
use strict;
use warnings;
use Test::More tests => 6;
BEGIN { use_ok('cpp_enum') }
require_ok('cpp_enum');

my $f = cpp_enum::Foo->new();

is($f->{hola}, $cpp_enum::Foo::Hello);

$f->{hola} = $cpp_enum::Foo::Hi;
is($f->{hola}, $cpp_enum::Foo::Hi);

$f->{hola} = $cpp_enum::Foo::Hello;
is($f->{hola}, $cpp_enum::Foo::Hello);

$cpp_enum::hi = $cpp_enum::Hello;

is($cpp_enum::hi, $cpp_enum::Hello);

File: Examples/test-suite/ruby/cpp_enum_runme.rb
#!/usr/bin/env ruby
#
# This test implementation is directly derived from its Python counterpart.
#

require 'swig_assert'
require 'cpp_enum'

f = Cpp_enum::Foo.new
swig_assert_equal('f.hola', 'Cpp_enum::Foo::Hello', binding)

f.hola = Cpp_enum::Foo::Hi
swig_assert_equal('f.hola', 'Cpp_enum::Foo::Hi', binding)

f.hola = Cpp_enum::Foo::Hello
swig_assert_equal('f.hola', 'Cpp_enum::Foo::Hello', binding)

Cpp_enum::hi = Cpp_enum::Hello
swig_assert_equal('Cpp_enum::hi', 'Cpp_enum::Hello', binding)
